thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -244,6 +246,15 @@ private void storeData(AnomalyDetectionOutputContext anomalyDetectionOutputConte
     String metricName = anomalyFunction.getSpec().getTopicMetric();
     MetricTimeSeries metricTimeSeries = anomalyDetectionInputContext.getDimensionKeyMetricTimeSeriesMap().get(dimensionMap);
 
+    /*
+    Check if current task is running offline analysis
+     */
+    boolean isOffline = false;
+    String jobName = DAO_REGISTRY.getJobDAO().getJobNameByJobId(jobExecutionId);
+    if (jobName != null && jobName.toLowerCase().startsWith(OFFLINE_PREFIX)) {
```

Comment:
It would be better to get properties of the detection task such as OFFLINE, BACKFILL from the DetectionTaskInfo. Using the job's name to pass on this information isn't such a good idea. The DetectionJobContext and DetectionTaskInfo were created to exactly this kind of information transfer from the creator to the executor. The tasks should be able to run, no matter what the job is named, purely on the basis of the task info and task context. Having booleans like offline and backfill (with predefined defaults), will act better to keep the platform generic, and not make the job property drive the task execution.